plugins/woocommerce-blocks/assets/js/blocks/product-collection/collections/register-product-collection.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import { BlockVariation, registerBlockVariation } from '@wordpress/blocks';
+import { addFilter } from '@wordpress/hooks';
+import { EditorBlock } from '@woocommerce/types';
+import { type ElementType } from '@wordpress/element';
+import type { BlockEditProps } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import blockJson from '../block.json';
+import {
+	SetPreviewState,
+	PreviewState,
+	ProductCollectionAttributes,
+} from '../types';
+
+// Extends BlockVariation to include an optional setPreviewState function for preview management.
+export interface ProductCollectionConfig extends BlockVariation {
+	preview?: {
+		setPreviewState?: SetPreviewState;
+		initialState?: PreviewState;
```

Comment:
```suggestion
		initialPreviewState?: PreviewState;
```

Considering the `setPreviewState` function has `Preview` in the middle I'd add the same to `initialState` for symmetry. Especially that later it's used like this:

```
	registerBlockVariation( blockJson.name, {
		...blockVariationArgs,
		attributes: {
			...blockVariationArgs.attributes,
			previewState: initialState, // I think I'd like to see previewState: initialPreviewState here
		},
	} );
```

Reply:
Made the changes in https://github.com/woocommerce/woocommerce/pull/45703/commits/9992019df1d3cb2f827127c76b81249650d7f5d2